src/Main/Main.tsx

```diff
@@ -0,0 +1,84 @@
+/* eslint-disable jsx-a11y/control-has-associated-label */
+
+import { useLocation } from 'react-router-dom';
+import { CSSTransition, TransitionGroup } from 'react-transition-group';
+import { Todo } from '../utils/types/type';
+import { TodoItem } from './TodoItem';
+
+type Props = {
+  todos: Todo [],
+  setTodos: (todos: Todo[]) => void,
+  setIsProcessing: (ids: number[]) => void,
+  isProcessing: number[],
+};
+
+export const Main:React.FC<Props> = ({
+  todos, setTodos, setIsProcessing, isProcessing,
+}) => {
+  const location = useLocation();
+  const pathName = location.pathname.slice(1);
+  const handlerToggleAll = () => {
+    const toggleVector = !!todos.some(todo => todo.completed === false);
+
+    setIsProcessing(toggleVector
+      ? todos
+        .filter(todo => todo.completed !== toggleVector)
+        .map(todo => todo.id)
+      : todos.map(todo => todo.id));
+
+    setTodos(todos.map(todo => {
+      return {
+        ...todo,
+        completed: toggleVector,
+      };
+    }));
+  };
+
+  const filteredTodos = () => {
```

Comment:
Recomputing filtered todos every time the component re-renders might impact performance if there are many items. Consider using the useMemo hook to memoize the result.

Reply:
in this case there is a constant update in any case.